functions/Get-DbaInstallDate.ps1

```diff
@@ -0,0 +1,188 @@
+                    } else 
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername 
+                    }					
+				}
+				catch 
+				{
+					Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance					
+				}
+
+				if ( $server.VersionMajor -ge 9 )
+				{ 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" -Silent $false 
+					$sql = "SELECT create_date FROM sys.server_principals WHERE sid = 0x010100000000000512000000"
+					$sqlInstallDate = $server.databases['master'].ExecuteWithResults($sql).tables
+					$sqlInstallDate = ($sqlInstallDate.rows).create_date.toString('MM-dd-yyyy') 
+
+				} else { 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" -Silent $false 
+					$sql = "SELECT schemadate FROM sysservers"
+					$sqlInstallDate = $server.databases['master'].ExecuteWithResults($sql).tables					
+					$sqlInstallDate = ($sqlInstallDate.rows).create_date.toString('MM-dd-yyyy') 
+				}											
+
+			} 
+
+			if ($SqlOnly -ne $true)			
+			{ 
+				Write-Message -Level Verbose -Message "Getting Windows Server Name for: $servername" -Silent $false 
+				$WindowsServerName = (Resolve-DbaNetworkName $servername -Credential $WindowsCredential).ComputerName
+				
+				try
+				{
+					Write-Message -Level Verbose -Message "Getting Windows Install date via CIM for: $WindowsServerName" -Silent $false 
+					$windowsInstallDate = (Get-CimInstance -ClassName win32_operatingsystem -ComputerName $windowsServerName -ErrorAction SilentlyContinue).InstallDate	
+					$windowsInstallDate = $windowsInstallDate.toString('MM-dd-yyyy')				
+				}
+				catch
+				{
+					try
+					{	
+						Write-Message -Level Verbose -Message "Getting Windows Install date via DCOM for: $WindowsServerName" -Silent $false 
+						$CimOption = New-CimSessionOption -Protocol DCOM
+						$CimSession = New-CimSession -Credential:$WindowsCredential -ComputerName $WindowsServerName -SessionOption $CimOption
+						$windowsInstallDate = ($CimSession | Get-CimInstance -ClassName Win32_OperatingSystem).LastBootUpTime
+
+						$windowsInstallDate = $windowsInstallDate.toString('MM-dd-yyyy')						
+					}
+					catch
+					{
+						Stop-Function -Message "Failed to connect to: $WindowsServerName" -Continue -Target $instance						
+					}
+				}
+			}
+
+			if ($sqlOnly -eq $true)
+			{ 
+				[PSCustomObject]@{
+					ComputerName = $server.NetName
+					InstanceName = $server.ServiceName
+					SqlServer = $server.Name
+					SqlInstallDate = $sqlInstallDate
+				}
+
+			} elseif ($WindowsOnly -eq $true)
+			{ 
+				[PSCustomObject]@{
+					ComputerName = $WindowsServerName										
+					WindowsInstallDate = $windowsInstallDate
+				}
+			} else 
+			{ 
+				[PSCustomObject]@{
+					ComputerName = $server.NetName
```

Comment:
To clean this up a bit you can make the one object with all the information and utilize something like this:

```powershell
$object = [PSCustomObject]@{
ComputerName = $server.NetName
InstanceName = $server.ServiceName
SqlServer = $server.InstanceName
SqlInstallDate = $sqlInstallDate
WindowsInstallDate = $windowsInstallDate
}
if ($sqlOnly) {
   Select-DefaultView -InputObject $object -Property ComputerName, InstanceName, SqlServer, SqlInstallDate
}
elseif ($WindowsOnly) {
   Select-DefaultView -InputObject $object -Property ComptuerName, WindowsInstallDate
}
else {
   Select-DefaultView -InputObject $object -Property ComputerName, InstanceName, SqlServer, SqlInstallDate, WindowsInstallDate
}
```

Have not tested but that should work.